core/domain/blog_services.py

```diff
@@ -637,8 +642,11 @@ def generate_url_fragment(title: str, blog_post_id: str) -> str:
     Returns:
         str. The url fragment of the blog post.
     """
-    lower_title = title.lower().replace(':', '')
-    hyphenated_title = lower_title.replace('  ', ' ').replace(' ', '-')
+    lower_title = title.lower()
+    # Removing special characters from url fragment.
+    simple_title = re.sub(r'[^a-zA-Z0-9 ]', '', lower_title)
+    hyphenated_title = (
+        simple_title.replace('-', ' ').replace('  ', ' ').replace(' ', '-'))
```

Comment:
Can you explain the logic behind this?

Reply:
Changed the logic a bit, Need to replace multiple space or hyphens with a single hyphen between words to get the desired url

Reply:
But why we need to do it this way? Why is there a possibility for multiple spaces?

Reply:
Since it is a blog post title it might be that the user wants to have a title like: 'Oppia Volunteer - The Palestine Story' . (I don't remember the exact title due to which the need to have something like this arose during blog post migration). Now if we simply replace spaces with hypen there will be multiple hyphens in the generated string. Also in some titles, author did have 2 spaces between words.